app/api/admin_statistics_api/events.py

```diff
@@ -1,42 +1,37 @@
 from flask_rest_jsonapi import ResourceDetail
-from marshmallow_jsonapi.flask import Schema
-from marshmallow_jsonapi import fields
-from datetime import datetime
-import pytz
+from flask import jsonify, Blueprint
+from flask_jwt import jwt_required
 
-from app.api.helpers.utilities import dasherize
 from app.api.bootstrap import api
 from app.models import db
-from app.models.event import Event
 from app.api.data_layers.NoModelLayer import NoModelLayer
 from app.api.helpers.db import get_count
-
-
-class AdminStatisticsEventSchema(Schema):
-    """
-    Api schema
-    """
-    class Meta:
-        """
-        Meta class
-        """
-        type_ = 'admin-statistics-event'
-        self_view = 'v1.admin_statistics_event_detail'
-        inflect = dasherize
-
-    id = fields.String()
-    draft = fields.Method("events_draft_count")
-    published = fields.Method("events_published_count")
-    past = fields.Method("events_past_count")
-
-    def events_draft_count(self, obj):
-        return get_count(Event.query.filter_by(state='draft'))
-
-    def events_published_count(self, obj):
-        return get_count(Event.query.filter_by(state='published'))
-
-    def events_past_count(self, obj):
-        return get_count(Event.query.filter(Event.ends_at < datetime.now(pytz.utc)))
+from app.api.schema.admin_statistics_schema.events import AdminStatisticsEventSchema
+from app.models.event import Event
+from app.models.event_topic import EventTopic
+from app.models.event_type import EventType
+
+event_statistics = Blueprint('event_statistics', __name__, url_prefix='/v1/admin/statistics')
+
+
+@event_statistics.route('/event-topics', methods=['GET'])
+@jwt_required()
+def event_topic_count():
+    event_topic_counts = {}
+    for event_topic in EventTopic.query.all():
+        value = get_count(Event.query.filter_by(event_topic_id=event_topic.id))
+        event_topic_counts[event_topic.name] = value
+    return jsonify(event_topic_counts)
+
+
+@event_statistics.route('/event-types', methods=['GET'])
+@jwt_required()
+def event_types_count():
+    event_types_counts = {}
+    for event_type in EventType.query.all():
+        value = get_count(Event.query.filter_by(event_topic_id=event_type.id))
```

Comment:
@bhaveshAn like I have been telling from the very beginning .. Why couldn't you use Postgres's query to group and aggregate ? 

Here, if there are 20 event types, you will be making 21 queries to get this statistics when all you need is a single query. I'm not sure why this is hard to understand.

Reply:
@niranjan94 Using query:
```
SELECT event_topics.name, count(events.event_topic_id) FROM events, event_topics WHERE events.event_topic_id=event_topics.id GROUP BY event_topics.name;
```
I am only getting the event topic which has a event and not 0 when there isn't any event of a event topic.
Is something wrong with the query?

> Here, if there are 20 event types, you will be making 21 queries to get this statistics when all you need is a single query. I'm not sure why this is hard to understand.

Earlier I mean the focus is on to get the dynamic response. That's why it happened.

Reply:
Hi @bhaveshAn 

You should be using a `JOIN` query. Try this one. Should work.

```sql
SELECT event_types.id as event_type_id, event_types.name as event_type, COUNT(events.id) AS count
FROM event_types
LEFT JOIN events
ON events.event_type_id = event_types.id
GROUP BY event_types.id
```

Reply:
Hi @niranjan94 Please review now.